lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -1375,6 +1375,17 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
     }
     return b.create<arith::DivFOp>(loc, payloadArgs[0], payloadArgs[1]);
   }
+  if (auto pow = dyn_cast<AtenPowTensorScalarOp>(op)) {
+    if (!pow.getType()
+             .cast<ValueTensorType>()
+             .getDtype()
+             .isa<mlir::FloatType>()) {
+      pow.emitError("unimplemented: non-floating point dtype");
+      return nullptr;
+    }
+    Value expF32 = b.create<arith::TruncFOp>(loc, b.getF32Type(), operands[1]);
```

Comment:
Can you explain the reason for truncation?

Reply:
Torch.Float type is f64, math.pow need src and exponent date type is same, so cast exponent f64 -> f32.

Reply:
There's no guarantee that the input tensor is f32. I would instead use the helper

https://github.com/llvm/torch-mlir/blob/92ed64626b8e45f62ee0ae295b58f853a513b302/lib/Conversion/TorchToLinalg/TorchToLinalg.cpp#L1255

and do something like this to get the `dtype`

https://github.com/llvm/torch-mlir/blob/92ed64626b8e45f62ee0ae295b58f853a513b302/lib/Conversion/TorchToLinalg/TorchToLinalg.cpp#L1467

Reply:
THANKS! Code has updated